controllers/topolvm_controller.go

```diff
@@ -0,0 +1,325 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	cutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	controllerName = "topolvm-controller"
+)
+
+type topolvmController struct{}
+
+// topolvmController unit satisfies resourceManager interface
+var _ resourceManager = topolvmController{}
+
+func (c topolvmController) getName() string {
+	return controllerName
+}
+
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=create;update;delete;get;list;watch
+
+func (c topolvmController) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	// get the desired state of topolvm controller deployment
+	controllerDeployment := getControllerDeployment(lvmCluster)
+
+	result, err := cutil.CreateOrUpdate(ctx, r.Client, controllerDeployment, func() error { return nil })
+
+	// log the result based on current reconcile
+	switch result {
+	case cutil.OperationResultCreated:
+		r.Log.Info("csi controller", "operation", result, "name", controllerDeployment.Name)
+	case cutil.OperationResultUpdated:
+		r.Log.Info("csi controller", "operation", result, "name", controllerDeployment.Name)
+	case cutil.OperationResultNone:
+		r.Log.Info("csi controller", "operation", result, "name", controllerDeployment.Name)
+	default:
+		r.Log.Error(err, "csi controller reconcile failure", "name", controllerDeployment.Name)
+		return err
+	}
+
```

Comment:
Please replace this with 
	if err != nil {
		r.Log.Error(err, "reconcile failure", "name", controllerDeployment.Name)
	} else {
		r.Log.Info("csi controller", "operation", result, "name", controllerDeployment.Name)
	}